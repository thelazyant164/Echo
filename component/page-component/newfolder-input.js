import { React, useState, useEffect } from 'react';
import { StyleSheet, Text, View, Modal, Pressable } from 'react-native';
// Requires Expo-compatible RNFS -> to be done last

export const FolderInput = ({ setShowModal }) =>{
  const [activeFolderName, setActiveFolderName] = useState(null);
  // useEffect(() => {
  //     setActiveFolderName(RNFS.DocumentDirectoryPath); // Run on first render to identify platform-specific directory name
  // }, [])

  const [files, setFiles] = useState([]);

  // const getFileContent = async (path) => {
  //   const reader = await RNFS.readDir(path);
  //   setFiles(reader);
  // };

  // useEffect(() => {
  //   getFileContent(activeFolderName); // Read content of current active directory
  // }, [activeFolderName]);

  return (
    <View style={styles.centeredView}>
      <Modal
//        animationType="slide"
        transparent={true}
        onRequestClose={() => {
          Alert.alert("Modal has been closed.");
          setModalVisible(!modalVisible);
        }}
      >
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <Text style={styles.modalText}>Hello World!</Text>
            <Pressable
              style={[styles.button, styles.buttonClose]}
              onPress={() => setShowModal(false)}
            >
              <Text style={styles.textStyle}>Hide Modal</Text>
            </Pressable>
          </View>
        </View>
      </Modal>
    </View>
  )
}


const styles = StyleSheet.create({
    centeredView: {
      flex: 1,
      justifyContent: "center",
      alignItems: "center",
      marginTop: 22,
    },
    modalView: {
      margin: 20,
      backgroundColor: "white",
      borderRadius: 20,
      padding: 35,
      alignItems: "center",
      shadowColor: "#000",
      shadowOffset: {
        width: 0,
        height: 2
      },
      shadowOpacity: 0.25,
      shadowRadius: 4,
      elevation: 5
    },
    button: {
      borderRadius: 20,
      padding: 10,
      elevation: 2
    },
    buttonOpen: {
      backgroundColor: "#F194FF",
    },
    buttonClose: {
      backgroundColor: "#2196F3",
    },
    textStyle: {
      color: "white",
      fontWeight: "bold",
      textAlign: "center"
    },
    modalText: {
      marginBottom: 15,
      textAlign: "center"
    }
});