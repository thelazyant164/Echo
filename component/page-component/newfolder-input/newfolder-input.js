import { React, useState, useEffect } from 'react';
import {
  StyleSheet, Text, TextInput, View, Modal, Pressable, Alert,
} from 'react-native';
import * as FileSystem from 'expo-file-system';
import { useSelector, useDispatch } from 'react-redux';
import { hideModal } from '../../pages/filesStoragePage/files-slider';

const styles = StyleSheet.create({
  centeredView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 22,
  },
  modalView: {
    position: 'absolute',
    width: '80%',
    height: '50%',
    zIndex: 1000,
    margin: 20,
    backgroundColor: 'white',
    borderRadius: 20,
    padding: 35,
    alignItems: 'center',
    shadowColor: '#000',
    shadowOffset: {
      width: 0,
      height: 2,
    },
    shadowOpacity: 0.25,
    shadowRadius: 4,
    elevation: 5,
  },
  button: {
    marginTop: 20,
    marginBottom: 20,
    borderRadius: 20,
    padding: 10,
    elevation: 2,
  },
  buttonOpen: {
    backgroundColor: '#F194FF',
    width: 200,
  },
  buttonClose: {
    backgroundColor: '#2196F3',
    width: 200,
  },
  textStyle: {
    color: 'white',
    fontWeight: 'bold',
    textAlign: 'center',
  },
  modalText: {
    marginBottom: 15,
    textAlign: 'center',
  },
  input: {
    borderColor: 'gray',
    width: 250,
    borderWidth: 1,
    borderRadius: 10,
    padding: 10,
    marginTop: 10,
    marginBottom: 10,
  },
});

export function FolderInput({ setShowModal }) {
  const [newDirectory, setNewDirectory] = useState('New folder');
  // Create new folder in current active directory
  const dispatch = useDispatch();

  const createNewFolder = async () => {
    await FileSystem.makeDirectoryAsync(
      `${FileSystem.documentDirectory}/${newDirectory}`,
      { intermediates: true },
    );
  };

  return (
    <View style={styles.centeredView}>
      <Modal
        animationType="slide"
        transparent
        onRequestClose={() => {
          Alert.alert('Modal has been closed.');
        }}
      >
        <View style={styles.centeredView}>
          <View style={styles.modalView}>
            <Text style={styles.modalText}>Create new folder</Text>
            <TextInput
              style={styles.input}
              onChangeText={setNewDirectory}
              value={newDirectory}
              placeholder="Folder name"
            />
            <Pressable
              style={[styles.button, styles.buttonClose]}
              onPress={() => {
                createNewFolder()
                  // .then(setActiveDirectory(newDirectory))
                  .then(dispatch(hideModal()))
                  .catch(({ message }) => { Alert.alert(message); });
              }}
            >
              <Text style={styles.textStyle}>Save</Text>
            </Pressable>
            <Pressable
              style={[styles.button, styles.buttonClose]}
              onPress={() => {
                dispatch(hideModal());
              }}
            >
              <Text style={styles.textStyle}>Cancel</Text>
            </Pressable>
          </View>
        </View>
      </Modal>
    </View>
  );
}
