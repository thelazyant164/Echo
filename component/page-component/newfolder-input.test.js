import { React } from 'react';
import renderer from 'react-test-renderer';
import { FolderInput } from './newfolder-input';

describe('<FolderInput />', () => {
  let newFolderInput;
  beforeEach(() => {
    newFolderInput = renderer.create(<FolderInput />).toJSON();
  });

  test('Renders correctly', () => {
    expect(newFolderInput).toMatchSnapshot();
  });
});
