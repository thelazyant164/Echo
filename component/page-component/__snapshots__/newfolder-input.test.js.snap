// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FolderInput /> Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "margin": 20,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
          }
        }
      >
        <Text
          style={
            Object {
              "marginBottom": 15,
              "textAlign": "center",
            }
          }
        >
          Hello World!
        </Text>
        <TextInput
          allowFontScaling={true}
          onSubmitEditing={[Function]}
          placeholder="Name your new export folder..."
          rejectResponderTermination={true}
          style={
            Object {
              "borderWidth": 1,
              "height": 20,
              "margin": 15,
              "padding": 10,
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#2196F3",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Hide Modal
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`;
